pkg/builder/parameters.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+// collectSharedParameters finds those parameters that show up often and hence can be
+// shared across all the paths to save space.
+func collectSharedParameters(sp *spec.Swagger) (namesByJSON map[string]string, ret map[string]spec.Parameter, err error) {
+
+	if sp == nil || sp.Paths == nil {
+		return nil, nil, nil
+	}
+
+	countsByJSON := map[string]int{}
+	shared := map[string]spec.Parameter{}
+	var keys []string
+
+	collect := func(p *spec.Parameter) error {
+		if (p.In == "query" || p.In == "path") && p.Name == "name" {
```

Comment:
Low count does not work for CRDs. I had that before in an early version. But for CRDs we only create the schema for one CRD at a time.

Don't think this is magic because this code here is highly specific for our endpoints. It will break in other ways if you apply it to generic restful web services.

Reply:
Ah, I guess the parameter collection during merging won't be able to pick that up since the parameters are not replaced during the build? Ok sounds good